### PR TITLE
feat: Change Notice field comment format and unit-test to enforce they exist

### PR DESCRIPTION
**Pull request by bdferris-v2**
As part of the larger Notice documentation refactor (#1324), PR #1365 added support for extracting field documentation from Javadoc comments (aka /** */ comment style).  Critically, most of the field comments use // after I added them in bulk in PR #1346.  This PR switches the style to /** */.

It also adds a unit-test to enforce that each Notice field specifies a comment of this format.

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java [hunk 1]
```diff
@@ -96,26 +96,26 @@ public void validate(NoticeContainer noticeContainer) {
...
+    /** The faulty record's `shapes.shape_pt_sequence`. */
     private final int shapePtSequence;
 
-    // The row number from `shapes.txt` of the previous shape point.
+    /** The row number from `shapes.txt` of the previous shape point. */
     private final long prevCsvRowNumber;
 
     // Actual distance traveled along the shape from the first shape point to the previous shape
-    // point.
+    /** point. */
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java:115]:
```suggestion
    /** Actual distance traveled along the shape from the first shape point to the previous shape point. */
```

**Maybe it is too long for the formatting rule

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java [hunk 2]
```diff
@@ -96,26 +96,26 @@ public void validate(NoticeContainer noticeContainer) {
...
 
-    // The faulty record's `shapes.shape_pt_sequence`.
+    /** The faulty record's `shapes.shape_pt_sequence`. */
     private final int shapePtSequence;
 
-    // The row number from `shapes.txt` of the previous shape point.
+    /** The row number from `shapes.txt` of the previous shape point. */
     private final long prevCsvRowNumber;
 
     // Actual distance traveled along the shape from the first shape point to the previous shape
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java:114]: Remove this line if the suggestion below is applied.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java [hunk 3]
```diff
@@ -189,29 +189,29 @@ static class EqualShapeDistanceSameCoordinatesNotice extends ValidationNotice {
...
+    /** The previous shape point's `shape_dist_traveled` value. */
     private final double prevShapeDistTraveled;
 
-    // The previous record's `shapes.shape_pt_sequence`.
+    /** The previous record's `shapes.shape_pt_sequence`. */
     private final int prevShapePtSequence;
 
     // Actual distance traveled along the shape from the first shape point to the previous shape
-    // point.
+    /** point. */
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java:214]:
```suggestion
    /** Actual distance traveled along the shape from the first shape point to the previous shape point. */
```
** Maybe it is too long for the formatting tool.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 4]
```diff
@@ -81,26 +81,26 @@ public void validate(NoticeContainer noticeContainer) {
...
 
-    // The faulty record's `stop_times.stop_sequence`.
+    /** The faulty record's `stop_times.stop_sequence`. */
     private final int stopSequence;
 
-    // The row number from `stop_times.txt` of the previous stop time.
+    /** The row number from `stop_times.txt` of the previous stop time. */
     private final long prevCsvRowNumber;
 
     // Actual distance traveled along the shape from the first shape point to the previous stop
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:99]: Remove this line if suggestion below is applied.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 5]
```diff
@@ -81,26 +81,26 @@ public void validate(NoticeContainer noticeContainer) {
...
-    // time.
+    /** time. */
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:100]:
```suggestion
    /** Actual distance traveled along the shape from the first shape point to the previous stop time. */
```
** Maybe it is too long for the formatting rule.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/NoticeDocumentationTest.java [hunk 6]
```diff
@@ -71,6 +72,24 @@ public void testThatAllValidationNoticesAreDocumented() {
         .isEmpty();
   }
 
+  @Test
+  public void testThatNoticeFieldsAreDocumented() {
```
Comment (davidgamez) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/NoticeDocumentationTest.java:76]: 💪

**Review by davidgamez — CHANGES_REQUESTED** [review]

**Comment by bdferris-v2** [comment]
I incorporated all the suggestions.  Thanks!

**Review by davidgamez — APPROVED** [review]
🚀

**Pull request merged**